lexer/lexer.go

```diff
@@ -340,10 +337,41 @@ func (l *Lexer) prevChar(steps int) byte {
 
 func (l *Lexer) readIdentifier() string {
 	position := l.position
-	for isLetter(l.ch) {
+	for isIdentifier(l.ch) {
 		l.readChar()
 	}
-	return l.input[position:l.position]
+	return string(l.input[position:l.position])
+}
+
+func isIdentifier(ch rune) bool {
+	return !isSpecialRune(ch) && !isDigit(ch) && !isEmpty(ch)
+}
+
+// We will not treat the special runes as identifiers
+func isSpecialRune(ch rune) bool {
+	specialRune := []rune{
+		' ', '\t', '\n', '\r', // whitespace
+		'+', '-', '*', '/', '^', '%', // operator
+		'=', '!', '>', '<', // comparison operator
+		'&', '|', // logical operator
+		'.', ',', ':', '"', ';', // punctuation
+		'{', '}', '[', ']', '(', ')', // braces, brackets, parentheses
+		'#', '$', // others
+	}
+	for _, v := range specialRune {
+		if ch == v {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Same comment as above -- I think it's pretty "risky" to have this whitelist here, especially considering we have a list of special characters in [token.go](https://github.com/abs-lang/abs/blob/master/token/token.go#L17-L69) as well, so we're kind of duplicating the work here.